src/avrpart.c

```diff
@@ -533,64 +500,163 @@ void avr_mem_display(const char *prefix, FILE *f, const AVRMEM *m,
                      const AVRPART *p, int verbose) {
   static unsigned int prev_mem_offset;
   static int prev_mem_size;
-  int i, j;
-  char * optr;
+  const char *table_colum[] = {"Memory", "Size", "Pg size", "Offset"};
+  const char *table_padding = "-------------------------------";
+  static int m_char_max[4] = {0};
```

Comment:
The use of static variables in a library function is rarely justified. In applications outside AVRDUDE users might want to call this function with various parts. This messes up the alignment.

I realise this function used to have static variables before, but I am also sure this is not really needed. Which part has memories that have the same contents and are therefore not worth listing twice? This is what aliases cater for.

I would have this function do away with step-wise printing:
  - one loop over all memories to compute widths
  - another loop to print memories nearly
 
No static variables needed.